test/test_jit.py

```diff
@@ -6781,29 +6836,45 @@ def forward(self, x):
 
         self.checkTrace(TransformerNet(), (torch.rand(5, 3, 224, 224),))
 
+    @staticmethod
+    def _test_mnist(self, device, check_export_import=True):
+        # eval() is present because dropout makes this nondeterministic
+        self.checkTrace(MnistNet().to(device).eval(), (torch.rand(5, 1, 28, 28, device=device),),
+                        export_import=check_export_import)
+
     def test_mnist(self):
-        class Net(nn.Module):
-            def __init__(self):
-                super(Net, self).__init__()
-                self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
-                self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
-                self.conv2_drop = nn.Dropout2d()
-                self.fc1 = nn.Linear(320, 50)
-                self.fc2 = nn.Linear(50, 10)
+        self._test_mnist(self, device='cpu')
 
-            def forward(self, x):
-                x = F.relu(F.max_pool2d(self.conv1(x), 2))
-                x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
-                x = x.view(-1, 320)
-                x = F.relu(self.fc1(x))
-                x = F.dropout(x, training=self.training)
-                x = self.fc2(x)
-                return F.log_softmax(x, dim=1)
+    @unittest.skipIf(not RUN_CUDA, "no CUDA")
+    def test_mnist_cuda(self):
+        # XXX: export_import on CUDA modules doesn't work (#11480)
+        self._test_mnist(self, device='cuda', check_export_import=False)
 
-        # eval() is present because dropout makes this nondeterministic
-        self.checkTrace(Net().eval(), (torch.rand(5, 1, 28, 28),))
+    @unittest.skipIf(not RUN_CUDA, "no CUDA")
+    def test_mnist_training_leaks_no_memory_cuda(self):
```

Comment:
How long does this test run? Maybe we could shorten it to 5 iterations?

Reply:
It takes 7 seconds to run so it is not too much

Reply:
7 seconds is a lot if you consider that all JIT tests take a minute to run. I don't see why 40 would be much safer than 5

Reply:
Unit tests should ideally take on the order of milliseconds. We usually can't deliver the results this fast, because we're doing a bunch of expensive gradgradchecks, but there's no reason why this test has to be slow.